### PR TITLE
GP-275 | Transaction History

### PR DESCRIPTION
**Pull request by esyeng**
- added styled Table component to DashboardContent.jsx, displays transaction history from account info stored in firebase
- include a new file to .gitignore (I know I'm the only one who uses the vscode debugger but can we add launch.json to the .gitignore as a kindness?)

### CODE REVIEW

**Comment by ElianaTroper** [comment]
Getting an error on attempting to open the dashboard:

image

I'm guessing this is related to me using an account which has not associated entries, and that's causing a null someplace

**Review by ElianaTroper — CHANGES_REQUESTED** [review]
Fix ID'd issue @esyeng

**Review by ElianaTroper — CHANGES_REQUESTED** [review]
It's working locally without crashing - however, the buttons on the Manage CDP page are missing with these changes

**Comment by ElianaTroper** [comment]
Some other more UI type feedback
- it would make sense to use the CDP ID rather than the address (like the manage page).
- it would probably make sense to convert the timestamp to a users local time, remove the time zone, remove the seconds, and remove the day of the week to reclaim some space
- (More debateable) getting rid of the fees paid would probably be good - they're generally small and aren't the focus of the transaction history

**Comment by mccabe-david** [comment]
> Some other more UI type feedback
> 
> * it would make sense to use the CDP ID rather than the address (like the manage page).
> * it would probably make sense to convert the timestamp to a users local time, remove the time zone, remove the seconds, and remove the day of the week to reclaim some space
> * (More debateable) getting rid of the fees paid would probably be good - they're generally small and aren't the focus of the transaction history

I disagree about the fees. I think we should be transparent. But, if we need more space, I get it.

**Comment by esyeng** [comment]
@ElianaTroper 

- swapped address for id but kept formatted val for later
- changed timestamp to locale based date (mm/dd/yyyy)
- keeping fees for now?
- I have some additional edits I'd like to add which we can discuss next week (algo explorer link, liquidation price etc) but it should be set to merge; other tables are back to how they were

File: src/components/TransactionHistory.jsx [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+function mAlgosToAlgos(num) {
+  return num / 1000000
+}
+
+function mAlgosToAlgosFixed(num) {
+  return mAlgosToAlgos(num).toFixed(2)
+}
+
+export async function loadDbActionAndMetrics() {
```
Comment (mccabe-david) [inline src/components/TransactionHistory.jsx:28]: Should this be moved to Firebase.jsx? So, that all the DB read and write functions are grouped together?

Comment (esyeng) [reply src/components/TransactionHistory.jsx:28]: makes sense to me, should I go ahead and move it there ?

**Pull request merged**